src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/CounterfactualLogger.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Keeps all counterfactuals state in one place
+    /// </summary>
+    internal class CounterfactualLogger
```

Comment:
This class should be named based on what it does: logging a telemetry event once and enforcing that a single log occurs, even if called multiple times. The name CounterfactualLogger implies that counterfactual logging events should always be logged once which is not the case and we shouldn't have our code imply that. This class should be designed to log a telemetry event (possibly of a particular type) only once.

Reply:
Renamed to `TelemetryOnceEmitter` to represent what the class does.

TelemetryOnceEmitter is useful for counterfactual logging